bigquery/gcloud/aio/bigquery/bigquery.py

```diff
@@ -296,10 +361,8 @@ async def insert_via_load(
             timeout: int = 60,
             ignore_unknown_values: bool = False,
             schema_update_options: Optional[List[SchemaUpdateOption]] = None
-        ) -> Dict[str, Any]:
+        ) -> Job:
         """Loads entities from storage to BigQuery."""
-        if not source_uris:
```

Comment:
Hmm, I'm not sure we can remove this -- BQ errors when an empty list is passed in here. Maybe we have to early-return a faked job or have this method return `Optional[Job]`?

Reply:
Why isn't it ok to raise on empty list? That seems to be an API requirement, I don't see why the client should pretend the call was successful. If we return fake Job the client will probably fail on fetching the job's status, if we return `None` it's gonna trigger `KeyError` downstream. Both cases will just make debugging process longer.

Reply:
I guess it's mostly to "fix" an oddity of their API -- if you write a function which requests we "load zero rows of data", and we do nothing, should we succeed or fail? I'd argue that's a success; the API, though, throws an error since that case does not have validation on the Google side of things.

Basically, if we remove this than users need to do the additional work of capturing errors and introspecting into them to make sure the error wasn't actually a success.

Reply:
Agreed 100% on the fake job, though. I'm leaning towards the `Optional[Job]` case -- that shouldn't cause KeyErrors since that's what the type hints are for in the first place! Linters should prevent folks from attempting to use an `Optional[Job]` as if it were a `Job`.

Reply:
> I guess it's mostly to "fix" an oddity of their API -- if you write a function which requests we "load zero rows of data", and we do nothing, should we succeed or fail? I'd argue that's a success; the API, though, throws an error since that case does not have validation on the Google side of things.

But it does have validation:
```
{
  "error": {
    "code": 400,
    "message": "Load configuration must specify at least one source URI",
    "errors": [
      {
        "message": "Load configuration must specify at least one source URI",
        "domain": "global",
        "reason": "invalid"
      }
    ],
    "status": "INVALID_ARGUMENT"
  }
}
```
That's just how google defined their API. 

> Basically, if we remove this than users need to do the additional work of capturing errors and introspecting into them to make sure the error wasn't actually a success.

Well that sounds bad, but wouldn't it be a single `insert_via_load(source_uris) if source_uris` check?
I see a problem though: a user is required to have enough awareness to know that the list cannot be empty and we have no way to reflect this requirement in type hints. At the same time, `Optional[Job]` is not the best in explaining the situation either.

I dunno. Let me know if that changed you mind, if not I won't fight over google's schema.

Reply:
Hmm, yeah, I guess it's on the client either way; they'll either need to write:
```python
if source_uris:
    job = insert_via_load(source_uris)
    # 
```
or 
```python
job = insert_via_load(source_uris)
if job:
    # 
```

I guess it doesn't really matter as much as I thought haha. I think I'm still slightly more inclined towards `Optional[Job]` (just personally find it a bit clearer / more understandable here /shrug) but I don't have any strong opinions either way. If you do keep this check out, please make sure that this change is both highlighted in release notes and we make sure none of our own apps will run into issues here!

Reply:
Ok, I'll stick to schema then. Well that's gonna be fun to verify all our projects!